compiler/src/main/kotlin/io/spine/protodata/subscriber/Subscriber.kt

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2021, TeamDev. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.protodata.subscriber
+
+import com.google.common.collect.ImmutableSet
+import io.spine.base.EventMessage
+import io.spine.protodata.ReflectiveBuilder
+import io.spine.server.event.EventDispatcher
+import io.spine.server.type.EventClass
+import io.spine.server.type.EventEnvelope
+
+/**
+ * A subscriber to a compiler event.
+ *
+ * Instances of `Subscriber`s are created via reflection. It is required that the concrete classes
+ * have a `public` no-argument constructor.
+ */
+public abstract class Subscriber<E : EventMessage>(
```

Comment:
Initially, I thought compiler events were translated into Spine events. And the subscribers were Spine-powered `@Subscriber`s. Now I see you have chosen a different approach. I am not sure why this is so.

Perhaps, we could meet again and draw some things on a board?